src/pages/Chat/ChatMain.tsx

```diff
@@ -0,0 +1,136 @@
+import { useEffect, useRef, useState } from 'react';
+import MessageItem from '@components/MessageItem';
+import { ChatInterface } from '@interface/ChatInterface';
+import Button from '@components/Button';
+
+import angryEmotion from '@assets/emotion/angry.png';
+import awesomeEmotion from '@assets/emotion/awesome.png';
+import heartEmotion from '@assets/emotion/heart.png';
+import smileEmotion from '@assets/emotion/smile.png';
+import surpriseEmotion from '@assets/emotion/surprise.png';
+import useChatMessages from '@hooks/useChatMessages';
+
+// 메시지 날짜 변경 시 표기될 날짜 정보
+const formatDate = (dateString: string) => {
+  const date = new Date(dateString);
+  return date.toLocaleDateString('ko-KR', {
+    year: 'numeric',
+    month: 'numeric',
+    day: 'numeric',
+  });
+};
+
+export default function ChatMain(data: ChatInterface) {
+  const chatContainerRef = useRef<HTMLDivElement>(null);
+  const [selectedMessageId, setSelectedMessageId] = useState<string | null>(
+    null
+  );
+
+  const { updateMessageInStorage } = useChatMessages();
+
+  const friendId = data.id;
+  const [messages, setMessages] = useState(data.messages);
+
+  // 이모지반영되면 리렌더링
+  useEffect(() => {
+    setMessages(data.messages);
+  }, [data.messages]);
+
+  // 자동 스크롤
+  useEffect(() => {
+    if (chatContainerRef.current) {
+      chatContainerRef.current.scrollTop =
+        chatContainerRef.current.scrollHeight;
+    }
+  }, [messages]);
+
+  const handleDoubleClick = (messageId: string) => {
+    setSelectedMessageId(messageId);
+  };
```

Comment:
이모지 선택을 위해 메시지를 더블클릭하면 텍스트와 요소들이 파란색으로 드래그 되는 현상이 있더라구요! 저도 이번주에 이모지 기능을 구현하면서 똑같이 겪었던 문제인데 `event.preventDefault();`를 사용하여 해결했습니다. 혹시 도움되실까 해서 올려요~!!

```suggestion
const handleDoubleClick = (index: number, event: React.MouseEvent) => {
        event.preventDefault(); // 요 라인 추가
        setSelectedMessageId(messageId);
    };
```